backend/Api/Staffing/ReadModelFactory.cs

```diff
@@ -0,0 +1,185 @@
+using Api.Common;
+using Api.Organisation;
+using Core.DomainModels;
+using Core.Services;
+
+namespace Api.Staffing;
+
+public class ReadModelFactory
+{
+    private readonly StorageService _storageService;
+
+    public ReadModelFactory(StorageService storageService)
+    {
+        _storageService = storageService;
+    }
+
+    public List<ConsultantReadModel> GetConsultantReadModelsForWeeks(string orgUrlKey, List<Week> weeks)
+    {
+        var firstDayInScope = DateService.FirstDayOfWorkWeek(weeks.First());
+        var firstWorkDayOutOfScope = DateService.LastWorkDayOfWeek(weeks.Last()).AddDays(1);
+
+        return _storageService.LoadConsultants(orgUrlKey)
+            .Where(c => c.EndDate == null || c.EndDate > firstDayInScope)
+            .Where(c => c.StartDate == null || c.StartDate <= firstWorkDayOutOfScope)
+            .Select(consultant => MapToReadModelList(consultant, weeks))
+            .ToList();
+    }
+
+    public static ConsultantReadModel MapToReadModelList(
+        Consultant consultant,
+        List<Week> weekSet)
+    {
+        weekSet.Sort();
+
+        var detailedBookings = DetailedBookings(consultant, weekSet);
+
+        var hoursPrWorkday = consultant.Department.Organization.HoursPerWorkday;
+        var hoursPrWeek = hoursPrWorkday * 5;
+
+
+        var bookingSummary = weekSet.Select(week =>
+            GetBookedHours(week, detailedBookings, consultant)
+        ).ToList();
+
+        //isOccupied should not include offered or sellable time, as it's sometimes necessary to "double-book"
+        var isOccupied = bookingSummary.All(b =>
+            b.BookingModel.TotalBillable + b.BookingModel.TotalPlannedAbstences + b.BookingModel.TotalVacationHours +
+            b.BookingModel.TotalHolidayHours >= hoursPrWeek);
+
+        return new ConsultantReadModel(
+            consultant.Id, consultant.Name, consultant.Email,
```

Comment:
10 args. Maybe we should use the consultant obj/class as param instead to shorten it?

Reply:
Fixed! :)